Remove spacesOss plugin

jportner opened the pull request:
Partially resolves #104152.

- [x] Refactor the SpacesContext provider's fetched space data to make it usable by other consumers
- [x] Add CopyToSpaceFlyout to the Spaces plugin's public API contract
- [x] Change CopyToSpaceFlyout to work more efficiently by using space data that was loaded by the SpacesContext provider
- [x] Remove plugin dependencies on `spacesOss` plugin (from `dashboard` and `savedObjectsManagement` plugins)
  * Includes: change `savedObjectsManagement` plugin to register the Spaces columns/actions directly
- [x] Remove `spacesOss` plugin completely
  * Includes: moving all types to `spaces` plugin

jportner committed:
Refactor SpacesContext value's shareToSpacesDataPromise

This was originally built for only the "share to spaces" components to
consume. However, we now have a need for other consumers too. I renamed
it to "spacesDataPromise" and I changed the type for each entry,
removing the specific "cannotShareToSpace" flag in favor of a generic
"isAuthorizedForPurpose" function that can be used for any purpose.

jportner committed:
Add CopyToSpaceFlyout to spaces public API contract

jportner committed:
Change CopyToSpaceFlyout to use data from SpacesContext provider

Now uses spacesDataPromise instead of fetching all spaces each time the
flyout is opened.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/)
* Commit: 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108975/901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f243b0540d954a53b1b0371feb22c1cfd5593de9) | [after](https://github.com/elastic/kibana/commit/901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9) | diff |
| --- | --- | --- | --- |
| `spaces` | 265.2KB | 265.1KB | -151.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f243b0540d954a53b1b0371feb22c1cfd5593de9) | [after](https://github.com/elastic/kibana/commit/901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9) | diff |
| --- | --- | --- | --- |
| `spaces` | 42.3KB | 42.3KB | -21.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"}]}
PIPELINE-->

jportner committed:
Remove dependencies on spacesOss plugin

The dashboard and savedObjectsManagement plugins now both depend on the
spaces plugin directly!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/)
* Commit: 3baa6a79f8a4149fd31fca317f20180b23e867d7
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/execution/node/428/log)

### History
* :green_heart: [Build #146326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/) succeeded 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/","number":"146420","commit":"3baa6a79f8a4149fd31fca317f20180b23e867d7"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/)
* Commit: 4d07617125f8be508a10e65e4a78548ef47ddc6c
* Storybooks not built
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/execution/node/186/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/execution/node/197/log)
* [Error signal](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/execution/node/358/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: metrics for 4d07617125f8be508a10e65e4a78548ef47ddc6c were not reported

### History
* :broken_heart: [Build #146420](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/) failed 3baa6a79f8a4149fd31fca317f20180b23e867d7
* :green_heart: [Build #146326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/) succeeded 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/","number":"146420","commit":"3baa6a79f8a4149fd31fca317f20180b23e867d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/","number":"146480","commit":"4d07617125f8be508a10e65e4a78548ef47ddc6c"}]}
PIPELINE-->

jportner committed:
Remove spacesOss plugin

jportner committed:
Remove spaces dependency on savedObjectsManagement

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/)
* Commit: c6cbe45eb839b9e7179001ccb7fe38c869da9ddd
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/execution/node/348/log)

### History
* :broken_heart: [Build #146480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/) failed 4d07617125f8be508a10e65e4a78548ef47ddc6c
* :broken_heart: [Build #146420](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/) failed 3baa6a79f8a4149fd31fca317f20180b23e867d7
* :green_heart: [Build #146326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/) succeeded 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/","number":"146420","commit":"3baa6a79f8a4149fd31fca317f20180b23e867d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/","number":"146480","commit":"4d07617125f8be508a10e65e4a78548ef47ddc6c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/","number":"146743","commit":"c6cbe45eb839b9e7179001ccb7fe38c869da9ddd"}]}
PIPELINE-->

jportner committed:
Fix ts refs...?

mgiota committed:
[RAC] [Observability] Use simpler alert severity level mapping (#109068)

* [RAC][Observability] remove severity fields from mapping keep only ALERT_SEVERITY

* temporarily remove severity value occurences

* remove ALERT_SEVERITY_VALUE occurences, this value is not being read and shown in the Observability alerts table

* remove duplicate ALERT_SEVERITY identifier

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

lukeelmers committed:
Add deprecation notice to dashboard import/export docs. (#108826)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146765/)
* Commit: 1758d1268099a866a4084351ce1fa4894894420b
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146765/execution/node/407/log)

### History
* :broken_heart: [Build #146743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/) failed c6cbe45eb839b9e7179001ccb7fe38c869da9ddd
* :broken_heart: [Build #146480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/) failed 4d07617125f8be508a10e65e4a78548ef47ddc6c
* :broken_heart: [Build #146420](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/) failed 3baa6a79f8a4149fd31fca317f20180b23e867d7
* :green_heart: [Build #146326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/) succeeded 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/","number":"146420","commit":"3baa6a79f8a4149fd31fca317f20180b23e867d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/","number":"146480","commit":"4d07617125f8be508a10e65e4a78548ef47ddc6c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/","number":"146743","commit":"c6cbe45eb839b9e7179001ccb7fe38c869da9ddd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146765/","number":"146765","commit":"1758d1268099a866a4084351ce1fa4894894420b"}]}
PIPELINE-->

scottybollinger committed:
[Workplace Search] Fix rendering bug on Private Source status (#109122)

* Fix rendering bug on Private Source status

* Fix rendering bug on Private Source status

* Refactor per code review

parkiino committed:
[Security Solution][Endpoint][Event Filters] Event filters updated UI (#109039)

nchaulet committed:
[Fleet] Missing migration backport in 7.14.0 (#109174)

jbudz committed:
Revert "[build] Remove optimize dir creation (#109014)"

This reverts commit c2f310006df2aff38db5bd8608bc76b55a73c4b4.

smith committed:
Increase width of service inventory sparkplot columns (#109137)

From 160 to 176px. Prevents overlapping of the labels.

Fixes #109103.

ymao1 committed:
[Alerting] Update rules detail page to resolve SO IDs if necessary (#108091)

* Adding internal resolve API to resolve rules given an ID

* Updating after merge

* Updating after merge

* Adding resolveRule api to client and adding spacesOss plugin dependency

* Handling 404 errors by calling resolve. Updating unit tests

* Handling aliasMatch and conflict results

* Fixing types

* Unit tests for spaces oss components

* Adding audit event for resolve

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

qn895 committed:
[ML] Add tooltip help text for Transform filter query control (#108720)

* [ML] Add tooltip help text

* [ML] Remove unused import

* [ML] Update tooltip content to be clearer

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

Co-authored-by: Lisa Cawley <lcawley@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
Fixed flakiness with visualize custom role (#109077)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
Fix flakiness in load and save agg-based visualizations (#109066)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kertal committed:
[Discover] Unskip discover tab field data functional tests (#107305)

pgayvallet committed:
fix tsref? bis

stratoula committed:
[TSVB] Long legend values support (#108023)

* [TSVB] Supports legends with long values

* Add a unit test

* Design optimization

* Revert changes

* Add the missing prop type

* Ensure that the limits are respected

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146830/)
* Commit: 98a3916e839e3e9f17be80751a5a6f5ea43d4932
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108975/98a3916e839e3e9f17be80751a5a6f5ea43d4932)
* [Documentation Changes](https://kibana_108975.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `savedObjectsManagement` | 79 | 82 | +3 |
| `spaces` | 240 | 237 | -3 |
| `spacesOss` | 4 | - | -4 |
| total |  |  | -4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `dashboard` | 221.4KB | 221.3KB | -172.0B |
| `savedObjectsManagement` | 140.8KB | 140.7KB | -81.0B |
| `spaces` | 265.2KB | 265.2KB | +3.0B |
| total |  |  | -250.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `savedObjectsManagement` | 34.3KB | 39.1KB | +4.8KB |
| `spaces` | 42.3KB | 35.8KB | -6.5KB |
| `spacesOss` | 2.5KB | - | -2.5KB |
| total |  |  | -4.3KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `savedObjectsManagement` | 96 | 0 | -96 |
| `spaces` | 110 | 198 | +88 |
| `spacesOss` | 77 | - | -77 |
| total |  |  | -85 |

#### API count missing comments
| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `savedObjectsManagement` | 85 | 0 | -85 |
| `spaces` | 4 | 16 | +12 |
| `spacesOss` | 10 | - | -10 |
| total |  |  | -83 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/dd9dd527187c3a1bafcfabb6db9af4057f9cd481) | [after](https://github.com/elastic/kibana/commit/98a3916e839e3e9f17be80751a5a6f5ea43d4932) | diff |
| --- | --- | --- | --- |
| `savedObjectsManagement` | 20 | 0 | -20 |


</details>

### History
* :broken_heart: [Build #146765](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146765/) failed 1758d1268099a866a4084351ce1fa4894894420b
* :broken_heart: [Build #146743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/) failed c6cbe45eb839b9e7179001ccb7fe38c869da9ddd
* :broken_heart: [Build #146480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/) failed 4d07617125f8be508a10e65e4a78548ef47ddc6c
* :broken_heart: [Build #146420](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/) failed 3baa6a79f8a4149fd31fca317f20180b23e867d7
* :green_heart: [Build #146326](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/) succeeded 901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146326/","number":"146326","commit":"901aa40e9de9fda4f4b9ce2497b6fa9231d97ad9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146420/","number":"146420","commit":"3baa6a79f8a4149fd31fca317f20180b23e867d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146480/","number":"146480","commit":"4d07617125f8be508a10e65e4a78548ef47ddc6c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146743/","number":"146743","commit":"c6cbe45eb839b9e7179001ccb7fe38c869da9ddd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146765/","number":"146765","commit":"1758d1268099a866a4084351ce1fa4894894420b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146830/","number":"146830","commit":"98a3916e839e3e9f17be80751a5a6f5ea43d4932"}]}
PIPELINE-->

azasypkin committed:
Introduce `Enroll` API endpoint. (#108835)

stratoula committed:
[XY, Pie] Long legend values support (#108365)

* [XY, Pie] Long legend values support

* Update vislib snapshots

* Fix truncate labels to work only for slice labels positioned outside the chart

* Address PR comments

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): moving @kbn/plugin-generator to babel transpiler (#109083)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

darnautov committed:
[ML] Fix date formatting in the alert context of the Anomaly detection job health rule type  (#109073)

* [ML] extend alert messages with job ids

* [ML] rename getJobsErrorMessages

* [ML] add rule name to a template

* [ML] utilize fieldFormats plugin for date formatting

* [ML] update types

* [ML] update unit tests

* [ML] guards for uiSettings and fieldRegistry, add custom errors

* [ML] add conditional call for uiSettings

* [ML] fix i18n

* [ML] tests for getCustomError

* [ML] fix i18n check

* [ML] create classes for custom errors

jportner commented on x-pack/plugins/maps/tsconfig.json:
```diff
@@ -16,9 +16,27 @@
   "references": [
     { "path": "../../../src/core/tsconfig.json" },
     { "path": "../../../src/plugins/maps_ems/tsconfig.json" },
+    { "path": "../../../src/plugins/dashboard/tsconfig.json" },
+    { "path": "../../../src/plugins/inspector/tsconfig.json" },
+    { "path": "../../../src/plugins/data/tsconfig.json" },
+    { "path": "../../../src/plugins/ui_actions/tsconfig.json" },
+    { "path": "../../../src/plugins/navigation/tsconfig.json" },
+    { "path": "../../../src/plugins/expressions/tsconfig.json" },
+    { "path": "../../../src/plugins/visualizations/tsconfig.json" },
+    { "path": "../../../src/plugins/embeddable/tsconfig.json" },
+    { "path": "../../../src/plugins/saved_objects/tsconfig.json" },
+    { "path": "../../../src/plugins/share/tsconfig.json" },
+    { "path": "../../../src/plugins/presentation_util/tsconfig.json" },
+    { "path": "../../../src/plugins/home/tsconfig.json" },
+    { "path": "../../../src/plugins/charts/tsconfig.json" },
+    { "path": "../../../src/plugins/usage_collection/tsconfig.json" },
+    { "path": "../../../src/plugins/kibana_react/tsconfig.json" },
+    { "path": "../../../src/plugins/kibana_utils/tsconfig.json" },
+
     { "path": "../features/tsconfig.json" },
     { "path": "../licensing/tsconfig.json" },
     { "path": "../file_upload/tsconfig.json" },
     { "path": "../saved_objects_tagging/tsconfig.json" },
+    { "path": "../security/tsconfig.json" }
```
@kibana-gis it seems these TS references were mistakenly omitted from your `tsconfig.json` file. Speaking with @spalger it seems there is a bug in the TS checker which did not surface until my other (unrelated) changes in this PR.

TL;DR these references should have always been here :+1:

jportner commented on x-pack/plugins/spaces/public/copy_saved_objects_to_space/lib/process_import_response.ts:
```diff
@@ -0,0 +1,46 @@
+  successfulImports: SavedObjectsImportSuccess[];
+}
+
+// This is derived from the function of the same name in the savedObjectsManagement plugin
+export function processImportResponse(
+  response: SavedObjectsImportResponse
+): ProcessedImportResponse {
+  const { success, errors = [], successResults = [] } = response;
+  const failedImports = errors.map<FailedImport>(({ error, ...obj }) => ({ obj, error }));
+  return {
+    success,
+    failedImports,
+    successfulImports: successResults,
+  };
+}
```
I needed to remove this to handle a cyclic dependency between the `spaces` and `savedObjectsManagement` plugin.

Since copy-to-space was implemented as a thin veneer over export/import, the CTS flyout used this existing `processImportResponse` function from the `savedObjectsManagement` plugin to process the import response.

Turns out we could pare this down quite a bit, we didn't need a lot of what the original function provided. So I decided to just recreate a CTS-specific function here to remove that problematic dependency :+1:

pgayvallet replied:
Seems like the most pragmatic approach

jportner commented on x-pack/plugins/spaces/public/spaces_context/wrapper_internal.tsx:
```diff
@@ -9,38 +9,38 @@ import type { PropsWithChildren } from 'react';
   const spaces = await spacesManager.getSpaces({ includeAuthorizedPurposes: true });
   const activeSpace = await spacesManager.getActiveSpace();
   const spacesMap = spaces
-    .map<ShareToSpaceTarget>(({ authorizedPurposes, disabledFeatures, ...space }) => {
+    .map<SpacesDataEntry>(({ authorizedPurposes, disabledFeatures, ...space }) => {
       const isActiveSpace = space.id === activeSpace.id;
-      const cannotShareToSpace = authorizedPurposes?.shareSavedObjectsIntoSpace === false;
       const isFeatureDisabled = feature !== undefined && disabledFeatures.includes(feature);
       return {
         ...space,
         ...(isActiveSpace && { isActiveSpace }),
-        ...(cannotShareToSpace && { cannotShareToSpace }),
         ...(isFeatureDisabled && { isFeatureDisabled }),
+        isAuthorizedForPurpose: (purpose: GetAllSpacesPurpose) =>
+          // If authorizedPurposes is not present, then Security is disabled; normally in a situation like this we would "fail-secure", but
+          // in this case we are dealing with an abstraction over the client-side UI capabilities. There is no chance for privilege
+          // escalation here, and the correct behavior is that if Security is disabled, the user is implicitly authorized to do everything.
+          authorizedPurposes ? authorizedPurposes[purpose] === true : true,
       };
     })
-    .reduce((acc, cur) => acc.set(cur.id, cur), new Map<string, ShareToSpaceTarget>());
+    .reduce((acc, cur) => acc.set(cur.id, cur), new Map<string, SpacesDataEntry>());
```
This function is run when the SpacesContext provider is first mounted. It fetches all spaces and passes them to consumers via the context `value`.

This was originally purpose-built for the share-to-space flyout, but I renamed it and added a generic `isAuthorizedForPurpose` function. I have two motivations behind this:
1. The CTS flyout can also reuse this data, so we don't need to fetch all spaces every time the flyout is rendered
2. Other consumers can reuse this data in the future so they don't have to fetch all spaces too

jportner reviewed (commented):
Author's notes for reviewers

jgowdyelastic reviewed (approved):
ML changes LGTM

afharo committed:
[Flaky-tests] Fix SOM edit-saved-objects test (#109215)

jportner committed:
Merge branch 'master' into pr/jportner/108975

nreese reviewed (approved):
maps changes LGTM. Thanks for fixing our tsconfig.json file

jportner committed:
Remove spacesOss dependency from triggers_actions_ui plugin

jbudz reviewed (approved):
limits.yml LGTM

jportner committed:
Expose useSpaces function to external consumers

jportner commented:
Ugh, I merged upstream and it pulled in those changes. Sorry everyone for unnecessary pings, I'll close this PR and open a new one 😫
